File: src/pages/Client/services/UserApi.js
import axios from "axios";

const url = 'http://localhost:8080/user'
const createHeaders = () => {
    const token = localStorage.getItem('token');
    return token ? { Authorization: `Bearer ${token}` } : {};
}
export const orderDonation = async (formData) => {
  return await axios.post(`${url}/order-donation`, formData, { headers: createHeaders() });
};

export const getOrderDate = async () =>{
    return await axios.get(`${url}/get-order-date`, { headers: createHeaders() });
};
export const getOrderHistory = async () =>{
    return await axios.get(`${url}/list-order`, { headers: createHeaders() });
};
export const orderDonationReceiving = async (formData) => {
  return await axios.post(`${url}/order-receiving `, formData, { headers: createHeaders() });
};
export const getOrderReceiveHistory = async (userId) =>{
    return await axios.get(`${url}/list-order/receive`, { headers: createHeaders() });
};
export const getBloodType = async () => {
  return await axios.get(`${url}/list-bloods`, { headers: createHeaders() });
}

export const guestGetBloodValidBags = async () => {
  return await axios.get(`${url}/list-blood-valid-bags`, { headers: createHeaders() });
}
export const getUrgentCanceledOrders = async () => {
  return await axios.get(`${url}/list-order-receive-urgent`, { headers: createHeaders() });
}

File: be/src/main/java/com/nonoru/superapp/controller/UserAPI.java
package com.nonoru.superapp.controller;

import com.nonoru.superapp.dto.request.OrderBloodDonationRequest;
import com.nonoru.superapp.dto.request.OrderBloodReceiveRequest;
import com.nonoru.superapp.dto.response.*;
import com.nonoru.superapp.entity.BloodType;
import com.nonoru.superapp.service.*;
//import com.nonoru.superapp.service.OrderBloodReceiveService;
import jakarta.validation.Valid;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/user")
@CrossOrigin(origins = "http://localhost:3000/")
public class UserAPI {
    @Autowired
    private OrderBloodDonationService bloodDonationService;
    @Autowired
    private OrderDateDonationService dateDonationService;
    @Autowired
    private UserService userService;
    @Autowired
    private BloodService bloodService;
    @Autowired
    private OrderBloodReceiveService bloodReceiveService;
    @Autowired
    private BloodBagService bloodBagService;


    @PostMapping("/order-donation")
    public ApiResponse<Void> createOrder
            (@Valid @RequestBody OrderBloodDonationRequest request) {
        bloodDonationService.createOrderBloodDonation(request);
        return ApiResponse.<Void>builder()
                .message("Tạo đơn hoàn tất! Vui lòng chờ để được xét duyệt")
                .build();
    }
    @GetMapping("/get-order-date")
    public ApiResponse<List<OrderDateDonationResponse>> getOrderDate(){
        return ApiResponse.<List<OrderDateDonationResponse>>builder()
                .data(dateDonationService.getOrderDateDonation())
                .build();
    }
    @GetMapping("/list-order")
    public ApiResponse<List<UserOrderDonationResponse>> getListOrder(){
        return ApiResponse.<List<UserOrderDonationResponse>>builder()
                .data(userService.getOrderDonationOnlySelf())
                .build();
    }

    @GetMapping("/list-bloods")
    public ApiResponse<List<BloodType>> getBloodTypes(){
        return ApiResponse.<List<BloodType>>builder()
                .data(bloodService.getAllBloodTypes())
                .build();
    }
    @PostMapping("/order-receiving")
    public ApiResponse<Void> createOrderReceiving
            (@Valid @RequestBody OrderBloodReceiveRequest request) {
        bloodReceiveService.createOrder(request);
        return ApiResponse.<Void>builder()
                .message("Tạo đơn hoàn tất! Vui lòng chờ để được xét duyệt")
                .build();
    }
    @GetMapping("/list-order/receive")
    public ApiResponse<List<OrderBloodReceiveResponse>> getListOrderReceiveUser(){
        return ApiResponse.<List<OrderBloodReceiveResponse>>builder()
                .data(bloodReceiveService.getOrderForUser())
                .build();
    }
    @GetMapping("/list-blood-valid-bags")
    public ApiResponse<List<BloodBagResponse>> getBloodBagsValid(){
        return ApiResponse.<List<BloodBagResponse>>builder()
                .data(bloodBagService.getBloodBagsValid())
                .build();
    }
    @GetMapping("/list-order-receive-urgent")
    public ApiResponse<List<OrderReceiveUrgentPublicResponse>> getOrderReceiveUgent(){
        return ApiResponse.<List<OrderReceiveUrgentPublicResponse>>builder()
                .data(userService.getOrderReceiveUrgentPublic())
                .build();
    }
}
